lib/rex/proto/kerberos/pac/krb5_pac.rb

```diff
@@ -0,0 +1,406 @@
+
+    # @!attribute [rw] user_account_control
+    #   @return [Integer] Set of bit flags that represent information about this account
+    ndr_uint32 :user_account_control, initial_value: USER_NORMAL_ACCOUNT | USER_DONT_EXPIRE_PASSWORD
+
+    # @!attribute [rw] sub_auth_status
+    #   @return [Integer] Subauthentication package's status code
+    ndr_uint32 :sub_auth_status
+
+    # @!attribute [rw] last_successful_i_logon
+    #   @return [FileTime] User account's msDS-LastSuccessfulInteractiveLogonTime
+    ndr_file_time :last_successful_i_logon
+
+    # @!attribute [rw] last_failed_i_logon
+    #   @return [FileTime] User account's msDS-LastFailedInteractiveLogonTime
+    ndr_file_time :last_failed_i_logon
+
+    # @!attribute [rw] failed_i_logon_count
+    #   @return [Integer] User account's msDS-FailedInteractiveLogonCountAtLastSuccessfulLogon
+    ndr_uint32 :failed_i_logon_count
+
+    # @!attribute [rw] reserved_3
+    #   @return [Integer] This member is reserved
+    ndr_uint32 :reserved_3
+
+    # @!attribute [rw] sid_count
+    #   @return [Integer] Total number of SIDs present in the ExtraSids member
+    ndr_uint32 :sid_count
+
+    # @!attribute [rw] extra_sids_ptr
+    #   @return [Integer] A pointer to a list of KERB_SID_AND_ATTRIBUTES structures that contain a list of SIDs
+    #   corresponding to groups in domains other than the account domain to which the principal belongs
+    krb5_sid_and_attributes_ptr :extra_sids_ptr, length: :sid_count
+
+    # @!attribute [rw] resource_group_domain_sid_ptr
+    #   @return [Integer] Pointer to SID of the domain for the server whose resources the client is authenticating to
+    prpc_sid :resource_group_domain_sid_ptr # prpc_sid :resource_group_domain_sid_ptr
+
+    # @!attribute [rw] resource_group_count
+    #   @return [Integer] Number of resource group identifiers stored in ResourceGroupIds
+    ndr_uint32 :resource_group_count
+
+    # @!attribute [rw] resource_group_ids_ptr
+    #   @return [Integer] Pointer to list of GROUP_MEMBERSHIP structures that contain the RIDs and attributes of the
+    #   account's groups in the resource domain
+    pgroup_membership_array :resource_group_ids_ptr, type: [:group_membership, { byte_align: 4 }]
+
+    def group_ids=(group_ids)
+      self.group_memberships = group_ids.map do |id|
+        { relative_id: id, attributes: SE_GROUP_ALL }
+      end
+    end
+  end
+
+  class Krb5ValidationInfoPtr < Krb5ValidationInfo
+    default_parameters byte_align: 8
+    extend RubySMB::Dcerpc::Ndr::PointerClassPlugin
+  end
+
+  class Krb5LogonInformation < BinData::Record
+    endian :little
+    # @!attribute [r] ul_type
+    #   @return [Integer] Describes the type of data present
+    virtual :ul_type, value: 0x01
+
+    type_serialization1 :type_serialization1, data_length: -> { data.num_bytes }
+    krb5_validation_info_ptr :data
+  end
+
+  class Krb5PacElement < BinData::Choice
+    krb5_logon_information 0x00000001
```

Comment:
Let's replace the magic `0x00000001` references with a named constants - which can be used in the presenter logic too